src/io/stderr.rs

```diff
@@ -87,6 +99,33 @@ enum Operation {
     Flush(io::Result<()>),
 }
 
+impl Stderr {
+    /// Locks this handle to the standard error stream, returning a writable guard.
+    ///
+    /// The lock is released when the returned lock goes out of scope. The returned guard also implements the Write trait for writing data.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # fn main() -> std::io::Result<()> { async_std::task::block_on(async {
+    /// #
+    /// use async_std::io;
+    /// use async_std::prelude::*;
+    ///
+    /// let stderr = io::stderr();
+    /// let mut handle = stderr.lock().await;
+    ///
+    /// handle.write_all(b"hello world").await?;
+    /// #
+    /// # Ok(()) }) }
+    /// ```
+    pub async fn lock(&self) -> StderrLock<'static> {
```

Comment:
Ah dang, I just realized something I didn't spot before: we should probably mark the APIs introduced in this patch as "unstable". `async/await` is coming up real soon, and our 1.0 release is coming with it. By marking it as "unstable" we give ourselves some time to fix any bugs we might encounter.

Sorry for spotting this so late! -- Otherwise this PR is looking :100: :100: :100: